src/main/webapp/app/assessment/athena.service.ts

```diff
@@ -153,4 +158,61 @@ export class AthenaService {
             }),
         );
     }
+
+    /**
+     * Get feedback suggestions for the given modeling submission from Athena.
+     *
+     * @param exercise The exercise for which a submission is assessed
+     * @param submission The assessed submission
+     * @return observable that emits the feedback suggestions as Feedback objects with the "FeedbackSuggestion:" prefix
+     */
+    public getModelingFeedbackSuggestions(exercise: Exercise, submission: ModelingSubmission): Observable<Feedback[]> {
+        return this.getFeedbackSuggestions<ModelingFeedbackSuggestion>(exercise, submission.id!).pipe(
+            map((suggestions) => {
+                const referencedElementIDs = new Set();
+
+                const model: UMLModel | undefined = submission.model ? JSON.parse(submission.model) : undefined;
+
+                return suggestions.map((suggestion, index) => {
+                    const feedback = new Feedback();
+                    feedback.id = index;
+                    feedback.credits = suggestion.credits;
+                    feedback.positive = suggestion.credits >= 1;
+
+                    // Even though Athena can reference multiple elements for the same feedback item, Apollon can only
+                    // attach feedback to one element, so we select the first element ID mentioned. To ensure that not
+                    // more than one feedback item is attached to the same element, we additionally ensure that the
+                    // same element is only referenced once.
+                    const referenceId: string | undefined = suggestion.elementIds.filter((id) => !referencedElementIDs.has(id))[0];
+
+                    if (referenceId) {
+                        feedback.type = FeedbackType.AUTOMATIC;
+                        feedback.text = suggestion.description;
+
+                        feedback.referenceId = referenceId;
+
+                        referencedElementIDs.add(referenceId);
+
+                        if (model) {
+                            if (feedback.referenceId) {
+                                const element = findElement(model, feedback.referenceId);
+                                feedback.referenceType = element?.type;
+                                feedback.reference = `${element?.type}:${referenceId}`;
+                            }
+                        }
+                    } else {
+                        feedback.type = FeedbackType.MANUAL_UNREFERENCED;
+                        feedback.text = `${FEEDBACK_SUGGESTION_IDENTIFIER}${suggestion.title}`;
+                        feedback.detailText = suggestion.description;
+                    }
+
+                    // Load grading instruction from exercise, if available
+                    if (suggestion.structuredGradingInstructionId != undefined) {
```

Comment:
Is there a reason why you only added the `!= undefined` here and not for the other checks?

Reply:
Fair point - it was implemented this was as for text and programming exercises, the structuredGradingInstructionId is also checked for inequality to undefined - but I suppose this change be changed to a generic boolean check 👍